modules/test/playwright/tests/users-admin-web/usersAndOrganizations.spec.ts

```diff
@@ -1077,3 +1079,182 @@ test(
 		).toBeVisible();
 	}
 );
+
+test(
+	'Can search users in organizations',
+	{tag: '@LPD-50958'},
+	async ({apiHelpers, usersAndOrganizationsPage}) => {
+		const organization =
+			await apiHelpers.headlessAdminUser.postOrganization();
+
+		const userAccount =
+			await apiHelpers.headlessAdminUser.postUserAccount();
+
+		await apiHelpers.headlessAdminUser.assignUserToOrganizationByEmailAddress(
+			organization.id,
+			userAccount.emailAddress
+		);
+
+		apiHelpers.data.push({
+			id: `${organization.id}_${userAccount.emailAddress}`,
+			type: 'organizationUserAccountAssociation',
+		});
+
+		await usersAndOrganizationsPage.goToOrganizations();
+		await (
+			await usersAndOrganizationsPage.organizationsTableRowLink(
+				organization.name
+			)
+		).click();
+
+		await expect(
+			usersAndOrganizationsPage.organizationUsersTable
+		).toBeVisible();
+
+		await usersAndOrganizationsPage.usersSearchBar.fill(userAccount.name);
+		await usersAndOrganizationsPage.usersSearchBarButton.click();
+
+		await expect(
+			(
+				await usersAndOrganizationsPage.organizationUsersTableRow(
+					1,
+					userAccount.name,
+					true
+				)
+			).row
+		).toBeVisible();
+
+		await usersAndOrganizationsPage.usersSearchBar.fill('test');
+		await usersAndOrganizationsPage.usersSearchBarButton.click();
+
+		await expect(usersAndOrganizationsPage.noResultsMessage).toBeVisible();
+	}
+);
+
+test(
+	'Can assign multiple users to an organization',
+	{tag: '@LPD-50958'},
+	async ({apiHelpers, page, usersAndOrganizationsPage}) => {
+		const userAccounts: TUserAccount[] = [];
+
+		for (let i = 0; i < 5; i++) {
+			const user = await apiHelpers.headlessAdminUser.postUserAccount();
+			userAccounts.push(user);
+		}
+
+		const organization =
+			await apiHelpers.headlessAdminUser.postOrganization();
+
+		await usersAndOrganizationsPage.goToOrganizations();
+
+		await (
+			await usersAndOrganizationsPage.organizationActionsMenu(
+				organization.name
+			)
+		).click();
+
+		await usersAndOrganizationsPage.assignUsersMenuItem.click();
+
+		for (const user of userAccounts) {
+			await (
+				await usersAndOrganizationsPage.assignUsersCheckbox(user.name)
+			).check();
+		}
+
+		await usersAndOrganizationsPage.assignUsersDoneButton.click();
+
+		for (const user of userAccounts) {
+			apiHelpers.data.push({
+				id: `${organization.id}_${user.emailAddress}`,
+				type: 'organizationUserAccountAssociation',
+			});
+		}
+
+		await waitForAlert(page);
```

Comment:
move this before the for (because if we don't have the success message we can avoid to add the relationship)

and at this point you can just to a single loop where you do the push and the expect